api_examples.py

```diff
@@ -0,0 +1,70 @@
+import json
+
+
+class APIExampleMixin:
+    """A TestCase mixin that allows access to API by Example retrieval methods."""
+
+    SERVER_NAME = 'localhost:8000'
+
+    def create_request(self, *args, **kwargs):
+        return super().create_request(*args, SERVER_NAME=self.SERVER_NAME, **kwargs)
+
+    def api_example_json_file(self, url, method):
+        """
+        Return the API example at a location generated from the view URL and method.
+
+        All URL kwargs in the URL are given only by their names, no other characters.
+        For example:
+            for POSTing to a view at `/patients/<pk>`,
+            call self.api_example_json_file('/patients/pk', 'post')
+            and you'll get 'api-description/patients/pk/post.json'
+        """
+        return '{base}/api-description{url}/{method}.json'.format(
+            base=self.EXAMPLES_DIR,
```

Comment:
If you're going to do that, it's probably better to replace most of the path with `EXAMPLES_DIR` since otherwise you're just going to be copying and pasting `EXAMPLES_DIR=django.conf.settings.PARENT_DIR` into every test class.

The point of this method is that it generates your filenames based on the URL of the endpoint you're testing, which will change from method to method.  The intention is that you'll set up your API by Example JSON files to mirror the URL structure of the project, as is the case in the project we brought this from.  Those will all be in the same folder, which in this case is hardcoded as `PARENT_DIR/api-description` but could in theory be elsewhere.  Replacing that part with a setting sounds like a good idea, but forcing the subclasses to do it themselves over and over again is wasted effort.